modules/map/src/MAP_Types.f90

```diff
@@ -1730,144 +1695,104 @@ SUBROUTINE MAP_CopyOtherState( SrcOtherStateData, DstOtherStateData, CtrlCode, E
       CALL SetErrStat(ErrID_Fatal, 'Error allocating DstOtherStateData%Fz_anchor.', ErrStat, ErrMsg,RoutineName)
       RETURN
     END IF
-    DstOtherStateData%c_obj%Fz_anchor_Len = SIZE(DstOtherStateData%Fz_anchor)
-    IF (DstOtherStateData%c_obj%Fz_anchor_Len > 0) &
-          DstOtherStateData%c_obj%Fz_anchor = C_LOC( DstOtherStateData%Fz_anchor( i1_l ) )
+    DstOtherStateData%C_obj%Fz_anchor_Len = SIZE(DstOtherStateData%Fz_anchor)
+    IF (DstOtherStateData%C_obj%Fz_anchor_Len > 0) &
+          DstOtherStateData%C_obj%Fz_anchor = C_LOC( DstOtherStateData%Fz_anchor( i1_l ) )
   END IF
     DstOtherStateData%Fz_anchor = SrcOtherStateData%Fz_anchor
 ENDIF
  END SUBROUTINE MAP_CopyOtherState
 
- SUBROUTINE MAP_DestroyOtherState( OtherStateData, ErrStat, ErrMsg, DEALLOCATEpointers )
+ SUBROUTINE MAP_DestroyOtherState( OtherStateData, ErrStat, ErrMsg )
   TYPE(MAP_OtherStateType), INTENT(INOUT) :: OtherStateData
   INTEGER(IntKi),  INTENT(  OUT) :: ErrStat
   CHARACTER(*),    INTENT(  OUT) :: ErrMsg
-  LOGICAL,OPTIONAL,INTENT(IN   ) :: DEALLOCATEpointers
   
   INTEGER(IntKi)                 :: i, i1, i2, i3, i4, i5 
-  LOGICAL                        :: DEALLOCATEpointers_local
   INTEGER(IntKi)                 :: ErrStat2
   CHARACTER(ErrMsgLen)           :: ErrMsg2
   CHARACTER(*),    PARAMETER :: RoutineName = 'MAP_DestroyOtherState'
 
   ErrStat = ErrID_None
   ErrMsg  = ""
 
-  IF (PRESENT(DEALLOCATEpointers)) THEN
-     DEALLOCATEpointers_local = DEALLOCATEpointers
-  ELSE
-     DEALLOCATEpointers_local = .true.
-  END IF
-  
 IF (ASSOCIATED(OtherStateData%H)) THEN
- IF (DEALLOCATEpointers_local) &
   DEALLOCATE(OtherStateData%H)
-  OtherStateData%H => NULL()
```

Comment:
Do we still need this?

Reply:
I've seen some conflicting guidance on whether the `deallocate` statement also nullifies pointers or leaves the associated status undefined (as in initialization). [This comment](community.intel.com/t5/Intel-Fortran-Compiler/Question-about-deallocation-of-pointers/m-p/754651#M10138) from Intel's "Doctor Fortran" to the question `How to decide if a pointer has been allocated or pointed?` makes me think that we should include the statement that sets the pointer to NULL() afterwards.
![image](https://github.com/OpenFAST/openfast/assets/6108781/2b11ccca-f769-4de6-9197-7ce2b438d204)